boards/BMS/Src/App/states/App_DriveState.c

```diff
@@ -1,15 +1,87 @@
 #include "states/App_DriveState.h"
 
+#include "App_Imd.h"
 #include "App_SharedMacros.h"
 
+static void App_RunState_SetImdPeriodicCanSignals(
+    struct BmsCanTxInterface *can_tx,
+    struct Imd *              imd)
+{
+    App_CanTx_SetPeriodicSignal_SECONDS_SINCE_POWER_ON(
+        can_tx, App_Imd_GetSecondsSincePowerOn(imd));
+    App_CanTx_SetPeriodicSignal_FREQUENCY(can_tx, App_Imd_GetPwmFrequency(imd));
+    App_CanTx_SetPeriodicSignal_DUTY_CYCLE(
+        can_tx, App_Imd_GetPwmDutyCycle(imd));
+
+    const enum Imd_Condition condition = App_Imd_GetCondition(imd);
+    App_CanTx_SetPeriodicSignal_CONDITION(can_tx, condition);
+
+    const struct Imd_PwmEncoding pwm_encoding = App_Imd_GetPwmEncoding(imd);
+
+    App_CanTx_SetPeriodicSignal_VALID_DUTY_CYCLE(
+        can_tx, pwm_encoding.valid_duty_cycle);
+
+    switch (condition)
+    {
+        case IMD_SHORT_CIRCUIT:
+        {
+            App_CanTx_SetPeriodicSignal_DUMMY_0_HZ(can_tx, 0);
```

Comment:
Why `DUMMY` here?

Reply:
So this has more to do with the way DBC is written. 
```
BO_ 105 BMS_IMD: 8 BMS
SG_ Condition M : 0|3@1+ (1,0) [0|1] "" LOGGER
...
SG_ Dummy_0Hz m0 : 12|8@1+ (1,0) [0|1] "" LOGGER
SG_ Insulation_Measurement_DCP_10Hz m1 : 12|16@1+ (1,0) [0|1] "kOhm" LOGGER
SG_ Insulation_Measurement_DCP_20Hz m2 : 12|16@1+ (1,0) [0|1] "kOhm" LOGGER
SG_ Speed_Start_Status_30Hz m3 : 12|16@1+ (1,0) [0|1] "" LOGGER
SG_ Dummy_40Hz m4 : 12|8@1+ (1,0) [0|1] "" LOGGER
SG_ Dummy_50Hz m5 : 12|8@1+ (1,0) [0|1] "" LOGGER
SG_ Dummy_Invalid m6 : 12|8@1+ (1,0) [0|1] "" LOGGER
```
The first 3 bits serve as a multiplexer for what is represented in bit 12 to 19. Essentially, the first 3 bits are an enum for `10Hz`, `20Hz`,...,`50Hz`. Some frequency like 10Hz has an additional payload associated with it (i.e.  `Insulation_Measurement_DCP_10Hz`), but other like `0Hz` does not have an additional payload associated with it so I just leave a dummy.

You could argue that since this is a dummy periodic signal so I shouldn't even bother setting it, and this line is redundant. But I thought I would make the code more consistent and just explicitly set the `DUMMY` signal value. Hope that explains it clearly.

Reply:
Yeah I'm still not sure why we have this signal in the first place?

Reply:
Oh yeah for some reason I thought I must give them placeholder values, but now I realize that doesn't make anything. Removed the dummy signals entirely :sweat_smile: